### PR TITLE
Fix AreEqual conversion on Assembly types

### PR DESCRIPTION
**Pull request by jairbubbles**

### CODE REVIEW

File: Meziantou.FluentAssertionsAnalyzers.Tests/NUnit4ToFluentAssertionsAnalyzerUnitTests.cs [hunk 1]
```diff
@@ -847,4 +847,66 @@ public void MyTest()
...
+            """
+            using System.Reflection;
+            using FluentAssertions;
+            using NUnit.Framework.Legacy;
+
+            class Test
+            {
+                public void MyTest(Assembly assembly1, Assembly assembly2)
+                {
+                    assembly2.Should().BeSameAs(assembly1);
```
Comment (meziantou) [inline Meziantou.FluentAssertionsAnalyzers.Tests/NUnit4ToFluentAssertionsAnalyzerUnitTests.cs:876]: Are you sure of this change? It looks like you change the semantic here.

- `ClassicAssert.AreSame()` => `Should().BeSameAs()`
- `AreEqual` => `Should().Be()`

Comment (meziantou) [reply Meziantou.FluentAssertionsAnalyzers.Tests/NUnit4ToFluentAssertionsAnalyzerUnitTests.cs:876]: I haven't dig too much in NUnit, but it seems AreEqual has many comparers based on the actual type: https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs#L34-L55

Comment (jairbubbles) [reply Meziantou.FluentAssertionsAnalyzers.Tests/NUnit4ToFluentAssertionsAnalyzerUnitTests.cs:876]: Yeah you're right my change is wrong. The problem is specific to `Assembly` type.

In Fluent it uses:

```csharp
public class AssemblyAssertions : ReferenceTypeAssertions<Assembly, AssemblyAssertions>
```

And it does not inherit from `ObjectAssertions` that provides the `Be` method.

Comment (jairbubbles) [reply Meziantou.FluentAssertionsAnalyzers.Tests/NUnit4ToFluentAssertionsAnalyzerUnitTests.cs:876]: I updated the code to target only the Assembly type, let me know what you think.

File: Meziantou.FluentAssertionsAnalyzers/NunitAssertAnalyzerCodeFixProvider.cs [hunk 2]
```diff
@@ -192,6 +192,10 @@ private static async Task<Document> Rewrite(Document document, SyntaxNode nodeTo
                 {
                     result = rewrite.UsingShould(right, "Equal", ArgumentList(left, arguments.Skip(2)));
                 }
+                else if (leftType.Name == "Assembly" && leftType.ContainingNamespace.ToDisplayString() == "System.Reflection")
```
Comment (meziantou) [inline Meziantou.FluentAssertionsAnalyzers/NunitAssertAnalyzerCodeFixProvider.cs]: Can't you compare the type with an actual symbol?(`compilation.GetTypeByMetadataName("System.Reflection.Assembly")`)

Comment (jairbubbles) [reply Meziantou.FluentAssertionsAnalyzers/NunitAssertAnalyzerCodeFixProvider.cs]: Yep much better indeed.

**Pull request merged**